face_shazam/pca_recognition.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf-8 -*-
+import itertools
+import logging
+import math
+
+import numpy as np
+from sklearn.svm import LinearSVC
+
+_logger = logging.getLogger(__name__)
+
+
+class PCARecognizer:
+    """ Class representing an face recognizer using PCA.
+    """
+
+    def __init__(self, data_set, training_percentage):
+        """ Constructor.
+        Params:
+            data_set (dict): The dictionary holding each subject's list of images (represented as ndarrays).
+            training_percentage (float): The percentage of training elements.
+        """
+        # TODO: will the recognizer receive training and testing data? or just training?
+        # Validate data set
+        if data_set is None or not isinstance(data_set, dict) or not data_set:
+            raise ValueError("None, non-dictionary or empty subjects dictionary")
+        if filter(lambda subject: not isinstance(subject, str), data_set.keys()):
+            raise ValueError("The subjects must be represented as strings")
+        if filter(lambda images: not isinstance(images, list) or filter(lambda elem: not isinstance(elem, np.ndarray),
+                                                                        images), data_set.values()):
+            raise ValueError("Each subject must have a list of ndarrays attached to it")
+        images_per_subject = len(data_set.values()[0])
```

Comment:
Si nos sobra tiempo, esto podriamos meterlo en un for e ir agregando de a pasos al `separated_data_set_container ` . Como nice to have nomas.

Reply:
O sea, ir chequeando a medida que vamos agregando al data set?